File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        unordered_map<int,int>r;
        for (int i : target){
            r[i]++;
        }
        int maxm=*max_element(target.begin(),target.end());
        vector<string>res;
        for (int i=1;i<=maxm;i++){
            res.push_back("Push");
            if (r.find(i)==r.end()){
                res.push_back("Pop");
            }
        }
        return res;
    }
};

File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        List<Integer> r=new ArrayList<>();
        for (int i : target){
            r.add(i);
        }
        int maxm=Collections.max(r);
        List<String>res=new ArrayList<>();
        for (int i=1;i<=maxm;i++){
            res.add("Push");
            if (!r.contains(i)){
                res.add("Pop");
            }
        }
        return res;
    }
}